langwatch/src/optimization_studio/components/workflow/NewWorkflowModal.tsx

```diff
@@ -88,6 +143,54 @@ export const NewWorkflowModal = ({
                   }
                 />
               ))}
+              <Box
+                as="div"
+                borderWidth="2px"
+                borderRadius="md"
+                borderStyle="dashed"
+                cursor="pointer"
+                backgroundColor="white"
+                data-testid="new-workflow-card-import"
+                onClick={() => fileInputRef.current?.click()}
+                onDragOver={(e) => {
+                  e.preventDefault();
+                  setIsDragging(true);
+                }}
+                onDragLeave={() => setIsDragging(false)}
+                onDrop={(e) => {
+                  e.preventDefault();
+                  setIsDragging(false);
+                  const file = e.dataTransfer.files[0];
+                  if (file) handleFileUpload(file);
```

Comment:
consider moving this to a hook so it's easier to see all of the logic in one place.